src/query/api/v1/middleware/rewrite.go

```diff
@@ -117,8 +124,37 @@ func rewriteRangeDuration(
 	if err != nil {
 		return err
 	}
-
-	attrs, err := store.QueryStorageMetadataAttributes(ctx, params.start, params.end, fetchOpts)
+	startTime, endTime := params.start, params.end
+
+	// Using the prometheus engine in this way should be considered
+	// optional and best effort. Fall back to the frequently accurate logic
+	// of using the start and end time in the request
+	if opts.PrometheusEngineFn != nil {
+		lookback := opts.DefaultLookback
+		if params.isLookbackSet {
+			lookback = params.lookback
+		}
```

Comment:
The actual lookback may get changed here: https://github.com/m3db/m3/pull/3926/files#diff-f284760ffa1aa45c70f82bdf3d8982cc7328962c64802875e8cf7c5dd9c61784L147

So you should probably pull that code above your query bounds calculation.

Reply:
Moved some code around so that I can redo the time range logic if the lookback changes.